Fix instructions(steps) for conversions/from_str exercice

jfchevrette opened the pull request:
The steps currently tell the user to return an error but the tests check for panic.

This updates the instructions to make it clearer that if the extraction/parsing fails, a panic should be raised

AbdouSeck reviewed (changes requested):
Hi @jfchevrette, thanks for suggesting changes to this exercise. The changes you're adding will not work. The implementation of a trait should never panic. It should always return a `Result`.

The `#[should_panic]` directives you see in the test cases are there to make sure what when `.unwrap()` is used on an erroneous `Result`, the program should panic. The tests themselves can be updated to avoid the usage the `.unwrap()` method and to remove the somewhat confusing `#[should_panic]` directives. I am more than happy to review and approve this PR if you take that on.

Please let me know if you have any questions.

Thanks,
Abdou

jfchevrette commented:
I messed something up in my rebase. Apologies. Fixing.

jfchevrette commented:
@AbdouSeck Thank you for the detailed response and for being welcoming and patient with new contributors. This is my first contribution to the rust ecosystem and it is nice to see such kindness.

I have changed my PR following your guidance. I hope it meets expectations now. What I did was remove the `#[should_panic]` and replace the function body with an `assert!` and the tests now use `is_err()` to validate whether or not the returned value is an error.

I also changed the steps instructions a bit again. I mostly added clarification/hints at the newer tests which were added recently `trailing_comma` and `trailing_comma_and_some_string`

Please let me know what you think and advise if you feel there's additional changes needed. Thanks!

AbdouSeck reviewed (approved):
@jfchevrette the tests look great. Thanks for updating them and for adding clarifications to the instructions.

fmoko commented:
@all-contributors please add @jfchevrette for content and code
